Modules/itertoolsmodule.c

```diff
@@ -92,6 +92,13 @@ class itertools.pairwise "pairwiseobject *" "clinic_state()->pairwise_type"
 #undef clinic_state_by_cls
 #undef clinic_state
 
+#define FREE_AND_CLEAR(ptr)     \
+do {                            \
+    void *tmp = (void *)ptr;    \
+    ptr = NULL;                 \
+    PyMem_Free(tmp);            \
+} while (0)
+
```

Comment:
Can it be replaced with [`Py_CLEAR()`](https://docs.python.org/3/c-api/refcounting.html#c.Py_CLEAR)?

> Decrement the reference count for object o. ***The object may be NULL***, in which case the macro has no effect; otherwise the effect is the same as for Py_DECREF(), except that the argument is also set to NULL. The warning for Py_DECREF() does not apply with respect to the object passed because the macro carefully uses a temporary variable and ***sets the argument to NULL before decrementing its reference count***.

Reply:
Note: a memory area allocated by, for example, `PyMem_Malloc` is not a Python object; it does not have a reference count. You need to treat it as you would treat any other memory allocation in C code.

Reply:
And come to think of it, that wrapper would not have been needed anyway: The reason for `Py_CLEAR`'s existence is because of the race condition that can occur since `Py_DECREF` may end up executing arbitrary code; that is not the case for `PyMem_Free`.